Make definition of new user configurable

citizen428 opened the pull request:
## What type of PR is this? (check all applicable)

- [X] Refactor
- [X] Feature

## Description

This PR makes the duration a user is considered new configurable via `SiteConfig`. We use this to limit the number of articles and comments a new user can create within a certain timeframe (currently hardcoded to 5 minutes).

## Related Tickets & Documents

https://github.com/forem/rfcs/pull/85

## QA Instructions, Screenshots, Recordings

![image](https://user-images.githubusercontent.com/47985/111252671-39c45b80-8644-11eb-8329-e6e93d1f5442.png)

### UI accessibility concerns?

n/a

## Added tests?

- [X] Yes

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [X] I've updated the [Admin Guide](https://forem.gitbook.io/forem-admin-guide/)
- [X] I will share this change internally with the appropriate teams

citizen428 commented on app/decorators/user_decorator.rb:
```diff
@@ -106,4 +106,9 @@ def fully_banished?
     access_tokens.any?
   end
+
+  def considered_new?
```
I deliberated for a bit if I really already want to add this, but Peter already hinted at more rate limits of a similar nature and it did make the code in `CommentsController` and `Article::Creator` quite a bit nicer so I went for it.

rhymes reviewed (commented):
I might be reading the code wrongly but, do we validate for `0` in case the admin enters that value ? I'm not sure the admin would understand that `0` means "midnight of the current day"

citizen428 commented:
@rhymes Good point! I always found the `beginning_of_day` bit a bit unintuitive, so I removed it now and use `0` to mean that the new user rate limits have been disabled (also added this info to the admin UI):

```ruby
  def considered_new?
    min_days = SiteConfig.user_considered_new_days
    return false unless min_days.positive?

    created_at.after?(min_days.days.ago)
  end
```

rhymes reviewed (approved):
LGTM!

vaidehijoshi commented on app/lib/constants/site_config.rb:
```diff
@@ -339,6 +339,10 @@ module SiteConfig
         placeholder: ""
       },
+      user_considered_new_days: {
+        description: "The number of days a user is considered new. Use 0 to disable these checks.",
```
> Use 0 to disable these checks

I read this a few times and I have a feeling that this might not be clear to some less technical users (at least, I can imagine some users thinking, _"disable what check?"_). Maybe this could be:
```suggestion
        description: "The number of days a user is considered new. The default is 3 days, but you can disable this entirely by inputting 0.",
```

I don't _love_ my suggestion, so feel free to change it; just trying to think of ways that this user-facing text could potentially be misinterpreted or misunderstood by a Forem Creator 😅

vaidehijoshi reviewed (approved):
I left a tiny comment about the user-facing text, but the code itself looks great! 👍🏽
